fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogFragment.kt

```diff
@@ -40,15 +72,174 @@ class TranslationsDialogFragment : BottomSheetDialogFragment() {
+                )
+                .nestedScroll(rememberNestedScrollInteropConnection()),
+        ) {
+            content()
+        }
+    }
+
+    @Composable
+    private fun TranslationsAnimation(
+        translationsVisibility: Boolean,
+        density: Density,
+        translationsOptionsHeightDp: Dp,
+        content: @Composable AnimatedVisibilityScope.() -> Unit,
+    ) {
+        AnimatedVisibility(
+            visible = translationsVisibility,
+            enter = expandIn(
+                initialSize = {
+                    with(density) {
+                        IntSize(
+                            0,
+                            translationsOptionsHeightDp.roundToPx(),
+                        )
+                    }
+                },
+            ) + fadeIn(
+                animationSpec = tween(
+                    easing = LinearOutSlowInEasing,
+                ),
+            ) + slideInHorizontally(),
+        ) {
+            content()
+        }
+    }
+
+    @Composable
+    private fun TranslationsOptionsAnimation(
+        translationsVisibility: Boolean,
+        density: Density,
+        translationsHeightDp: Dp,
+        translationsWidthDp: Dp,
+        content: @Composable AnimatedVisibilityScope.() -> Unit,
+    ) {
+        AnimatedVisibility(
+            visible = translationsVisibility,
+            enter = expandIn(
+                animationSpec = tween(
+                    easing = LinearOutSlowInEasing,
+                ),
+                initialSize = {
+                    with(density) {
+                        IntSize(
+                            0,
+                            translationsHeightDp.roundToPx(),
+                        )
+                    }
+                },
+            ) + fadeIn(
+                animationSpec = tween(
+                    easing = LinearOutSlowInEasing,
+                ),
+            ) + slideInHorizontally(
+                initialOffsetX = { with(density) { translationsWidthDp.roundToPx() } },
+                animationSpec = tween(
+                    easing = LinearOutSlowInEasing,
+                ),
+            ),
+        ) {
+            content()
+        }
+    }
+
+    @Composable
+    private fun ShowTranslations(onDismiss: () -> Unit) {
+        TranslationsDialogBottomSheet(
+            onSettingClicked = onDismiss,
+        )
+    }
+
+    @Composable
+    private fun ShowTranslationOptions(onDismiss: () -> Unit) {
+        TranslationOptionsDialogBottomSheet(
+            translationOptionsList = getTranslationOptionsList(),
+            onBackClicked = onDismiss,
+            onTranslationSettingsClicked = {
+                findNavController().navigate(
+                    TranslationsDialogFragmentDirections.actionTranslationsDialogToTranslationSettingsFragment(),
+                )
+            },
+            aboutTranslationClicked = {},
+        )
+    }
```

Comment:
Let's extract all of these private @Composable functions into a separate file named `TranslationsDialogBottomSheet`.

Reply:
Done.